src/coreclr/tools/Common/TypeSystem/Ecma/EcmaField.cs

```diff
@@ -98,7 +101,11 @@ private TypeDesc InitializeFieldType()
             BlobReader signatureReader = metadataReader.GetBlobReader(metadataReader.GetFieldDefinition(_handle).Signature);
 
             EcmaSignatureParser parser = new EcmaSignatureParser(Module, signatureReader, NotFoundBehavior.Throw);
-            var fieldType = parser.ParseFieldSignature();
+            var fieldType = parser.ParseFieldSignature(out EmbeddedSignatureData[] data);
```

Comment:
It looks wasteful to use the expensive parsing method that creates `Stack`, etc. Can we have `bool` on the parser that tracks whether we have encountered any modifiers?

Reply:
(Or make the expensive parsing method cheaper, e.g. use inline storage for stack of limited depth.)

Reply:
The modifiers handling within the type system is utter garbage. The Stack should be rewritten to use a stack-based linked list once C# can express that (I haven't been keeping track with all the `unscoped ref` stuff - maybe it can already express it). I don't think it makes sense to add extra complexity to the already incomprehensible modifiers handling if we know we'll be able to get rid of this cost.

We already pay this cost when parsing method signatures. I've never seen it in profiles.